### PR TITLE
feat: refactor workspace commands using new api service

### PR DESCRIPTION
**Pull request by esasova**

### CODE REVIEW

File: Babylon/commands/api/workspaces/delete.py [hunk 1]
```diff
@@ -1,54 +1,57 @@
...
 @command()
 @wrapcontext()
-@pass_context
 @timing_decorator
+@retrieve_state
 @pass_azure_token("csm_api")
-@option("-D", "force_validation", is_flag=True, help="Force Delete")
-@argument("id", type=QueryType(), required=False)
-@inject_context_with_resource({"api": ['url', 'organization_id']})
+@option("-D", "force_validation", is_flag=True, default=True, help="Force Delete")
```
Comment (sellisd) [inline Babylon/commands/api/workspaces/delete.py:26]: typo: missing dashes (-) before the long option (`--force_validation`)
also it is easier to memorize if short and long command line arg have some letter in common, perhaps at some point we should have a global look at the commands and rethink if `D` and `force_validation` are the best options

Comment (sellisd) [reply Babylon/commands/api/workspaces/delete.py:26]: ❌ ignore the above 😃 I see it was indeed as the variable name

File: Babylon/commands/api/workspaces/create.py [hunk 2]
```diff
@@ -1,91 +1,89 @@
...
+    help="Workspace security role",
+)
+@option("--organization-id", type=str)
+@option("--name", "name", type=str)
+def create(
+    state: Any,
+    azure_token: str,
+    organization_id: str,
+    name: Optional[str],
+    payload: Optional[pathlib.Path] = None,
```
Comment (sellisd) [inline Babylon/commands/api/workspaces/create.py:53]: shouldn't payload be required when creating a workspace?

File: Babylon/commands/api/workspaces/service/api.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    def get_all(self):
+        response = oauth_request(
+            f"{self.url}/organizations/{self.organization_id}/workspaces",
+            self.azure_token,
+        )
+        return response
+
+    def get(self):
+        workspace_id = self.state["api"]["workspace_id"]
```
Comment (Leopold-Cramer) [inline Babylon/commands/api/workspaces/service/api.py:34]: is it really needed to put it here instead of in the init ?

**Review by Leopold-Cramer — APPROVED** [review]

File: Babylon/commands/api/workspaces/create.py (hunk 2 above)
Comment (esasova) [reply Babylon/commands/api/workspaces/create.py:53]: excellent question, I will find out :laughing:

File: Babylon/commands/api/workspaces/create.py
Comment (sellisd) [inline Babylon/commands/api/workspaces/create.py:1]: The state is not shared between users so we should 1) not mix swagger and babylon commands, 2) not have multiple people sharing resources?

File: Babylon/commands/api/workspaces/service/api.py (hunk 3 above)
Comment (esasova) [reply Babylon/commands/api/workspaces/service/api.py:34]: you are right, apparently, it doesn't break other commands, I'll change it ! :+1:

File: Babylon/commands/api/workspaces/delete.py [hunk 4]
```diff
@@ -1,54 +1,57 @@
...
     if response is None:
         return CommandResponse.fail()
-    logger.info(f"Successfully deleted workspace {workspace_id} from organization {context['api_organization_id']}")
+    if not workspace_id or workspace_id == state["services"]["api"]["workspace_id"]:
+        state["services"]["api"]["workspace_id"] = ""
+        env.store_state_in_local(state)
+        env.store_state_in_cloud(state)
+        logger.info(f"Workspace {service_state['api']['workspace_id']} has been successfully removed from the state")
+    else:
+        logger.info(f"Workspace {service_state['api']['workspace_id']} has been successfully deleted")
```
Comment (Leopold-Cramer) [inline Babylon/commands/api/workspaces/delete.py:56]: Can use logger.info(SUCCESS_DELETED("workspace", workspace_id)) instead

File: Babylon/commands/api/workspaces/delete.py [hunk 5]
```diff
@@ -1,54 +1,57 @@
...
+    workspace_service = WorkspaceService(state=service_state, azure_token=azure_token)
+    response = workspace_service.delete()
     if response is None:
         return CommandResponse.fail()
-    logger.info(f"Successfully deleted workspace {workspace_id} from organization {context['api_organization_id']}")
+    if not workspace_id or workspace_id == state["services"]["api"]["workspace_id"]:
+        state["services"]["api"]["workspace_id"] = ""
+        env.store_state_in_local(state)
+        env.store_state_in_cloud(state)
+        logger.info(f"Workspace {service_state['api']['workspace_id']} has been successfully removed from the state")
```
Comment (Leopold-Cramer) [inline Babylon/commands/api/workspaces/delete.py:54]: can use logger.info(SUCCESS_CONFIG_UPDATED("api", "workspace_id"))

**Pull request closed without merging**